sdk/identity/azure-identity/src/test/java/com/azure/identity/AzureCliCredentialTest.java

```diff
@@ -71,17 +69,18 @@ public void azureCliCredentialAzNotLogInException() throws Exception {
         TokenRequestContext request = new TokenRequestContext().addScopes("AzureNotLogin");
 
         // mock
-        IdentityClient identityClient = PowerMockito.mock(IdentityClient.class);
-        when(identityClient.authenticateWithAzureCli(request))
+        try (MockedConstruction<IdentityClient> identityClientMock = mockConstruction(IdentityClient.class, (identityClient, context) -> {
+            when(identityClient.authenticateWithAzureCli(request))
                 .thenReturn(Mono.error(new Exception("Azure not Login")));
-        when(identityClient.getIdentityClientOptions()).thenReturn(new IdentityClientOptions());
-        PowerMockito.whenNew(IdentityClient.class).withAnyArguments().thenReturn(identityClient);
-
-        // test
-        AzureCliCredential credential = new AzureCliCredentialBuilder().build();
-        StepVerifier.create(credential.getToken(request))
-            .expectErrorMatches(e -> e instanceof Exception && e.getMessage().contains("Azure not Login"))
-            .verify();
+            when(identityClient.getIdentityClientOptions()).thenReturn(new IdentityClientOptions());
+        })) {
+            // test
+            AzureCliCredential credential = new AzureCliCredentialBuilder().build();
+            StepVerifier.create(credential.getToken(request))
+                .expectErrorMatches(e -> e instanceof Exception && e.getMessage().contains("Azure not Login"))
+                .verify();
+            Assert.assertNotNull(identityClientMock);
```

Comment:
Was this for testing the change?

Reply:
The `assertNotNull`? This is to satisfy checkstyle. Due to the style of the mocks I rarely use the variable name, but there's no way to tell check-style it's fine and this is going to get cleaned up by the try-with-resources. This seemed like a reasonable way to fix it. I could suppress the warning if we think that's better, but I try not to suppress things if possible.